### PR TITLE
Devel

### PR DESCRIPTION
**Pull request by Galithil**
latest changes : improved piper modes, and this should also fix up the mes with the fasq files for the piper setup file

### CODE REVIEW

**Comment by vezzi** [comment]
looks good to me

File: ngi_pipeline/engines/piper_ngi/workflows.py [hunk 1]
```diff
@@ -122,10 +122,19 @@ def workflow_dna_variantcalling(qscripts_dir_path, setup_xml_path, global_config
...
+                     'format; should be list. Ignoring these parameters.')
+        else:
+            cl_string += " -jobNative {}".format(" ".join(job_native_args))
     if exec_mode == "sbatch":
         # Execute from within an sbatch file (run jobs on the local node)
-        num_threads = int(config.get("piper", {}).get("threads") or 8)
-        job_runner = "Shell"
+        num_threads = int(config.get("piper", {}).get("threads") or 16)
+        job_runner = config.get("piper", {}).get("shell_jobrunner") or \
+                     "ParallelShell --super_charge --ways_to_split 4 --job_scatter_gather_directory $SNIC_TMP/scatter_gather"
```
Comment (mariogiov) [inline ngi_pipeline/engines/piper_ngi/workflows.py]: Has anyone tested this? I was getting some samtools errors when I ran the tests but Johan ran with the same cl and it worked fine so we should probably triple-check. Just a heads up!

File: ngi_pipeline/engines/piper_ngi/launchers.py [hunk 2]
```diff
@@ -93,7 +93,11 @@ def analyze(project, sample, exec_mode="sbatch", restart_finished_jobs=False,
                                                             project_name=project.dirname,
                                                             project_id=project.project_id,
                                                             sample_id=sample.name)
-                setup_xml_cl, setup_xml_path = build_setup_xml(project=project,
+
+                #let's update the project to keep only the right fastq files
+                updated_project, src_aln_files, src_alnqc_files = \
+                    collect_files_for_sample_analysis(project, sample, restart_finished_jobs)
```
Comment (mariogiov) [inline ngi_pipeline/engines/piper_ngi/launchers.py]: This code gets run in the `sbatch_piper_sample` funciton below, also. I guess there's no harm but it's not really ideal? Also we should update the `genotype()` code below to do the same thing if this fixes a bug with the setup.xml generation.

**Comment by vezzi** [comment]
@mariogiov this is in devel so it is not yet in production. Before moving this to produciton we need to run a full test. Clearly a smaller test would make this things easyer :-D

**Comment by mariogiov** [comment]
@vezzi haha right, automated deployment and a test suite are numbers one and two on my to-do list -- I'll keep working on these this week and next.

**Pull request merged**